### ISSUE TITLE
libhiredis0.14, httpd, libcjose0 libapache2-mod-auth-openidc

### ISSUE THREAD

**Issue by fong1999** [redis/hiredis#1227]
Hi! I am reaching out is to ask if you have any other versions of libhiredis that we can use. 

We are having CVE with our httpd base image. When we upgrade to the latest version of httpd, the CVE is gone. However, our app will not start. The error we are getting is related to redis. From the list that we can find, **libhiredis0.14 (0.14.1-4)** is the only version we can see. So we had to use the earlier version - **httpd:2.4.56-debian-ff** which work with **libhiredis0.14**. Any suggestions? 

Here is what we have currently:

`apt install -y libcjose0 libapache2-mod-auth-openidc libhiredis0.14`

https://packages.debian.org/sid/libs/

https://packages.debian.org/sid/libhiredis0.14

Please let me know if you need more information.

Thank you,
Fong

**Comment by michael-grunder** [comment]
I'm not sure I fully understand the question.  There are several hiredis releases since v0.14 available on pretty much all package repositories.

That said, if you have a specific httpd extension or mod that can only talk with hiredis v0.14 then the only real solution would be to update the extension to use the newer version of hiredis.  

Looks like the offending package is`libapache2-mod-auth-openidc`?
```bash
❯ apt-cache show libapache2-mod-auth-openidc|grep Depends
Depends: libapr1 (>= 1.2.7), libaprutil1 (>= 1.4.0), libc6 (>= 2.14), libcjose0 (>= 0.4.1), libcurl4 (>= 7.16.2), libhiredis0.14 (>= 0.14.0), libjansson4 (>= 2.0.1), libpcre3, libssl3 (>= 3.0.0~~alpha1), apache2-api-20120211, apache2-bin (>= 2.4.16)
```

When I install that, it seems to specifically want hiredis v0.14.0.  This would have to be rectified in that package.

**Comment by fong1999** [comment]
@michael-grunder  Hi Michael, Thank you for your message!

Below is what we get too:

_❯ apt-cache show libapache2-mod-auth-openidc|grep Depends
Depends: libapr1 (>= 1.2.7), libaprutil1 (>= 1.4.0), libc6 (>= 2.14), libcjose0 (>= 0.4.1), libcurl4 (>= 7.16.2), libhiredis0.14 (>= 0.14.0), libjansson4 (>= 2.0.1), libpcre3, libssl3 (>= 3.0.0~~alpha1), apache2-api-20120211, apache2-bin (>= 2.4.16)_

It specifically wants hiredis v0.14.0. 

 Are you suggesting to upgrade library - _libapache2-mod-auth-openidc_ ? We tried to use different version of httpd other than httpd:2.4.56-debian-ff. But It doesn't work. 

_apt install -y libcjose0 libapache2-mod-auth-openidc_ 

I thought the above is to pull the latest version of libapache2-mod-auth-openidc... So should submit a question to libapache2-mod-auth-openidc and ask them? 

Thanks,
Fong

**Comment by michael-grunder** [comment]
> I thought the above is to pull the latest version of libapache2-mod-auth-openidc... So should submit a question to libapache2-mod-auth-openidc and ask them?

Unfortunately I don't really know anything about httpd mods or packages.  When I install the openidc mod on a fresh debian container, it installs hiredis v0.14.0 on it's own, so theoretically that should work just fine.

I'd need more information about what error message you're seeing in terms of app startup.  There are quite a few tricky edge cases around this stuff.

**Comment by fong1999** [comment]
@michael-grunder  Hi Michael, Thanks for your reply!

When we use the following combination in our Docker image below. Our app starts as normal.

```RUN npm run bundle

FROM /library/httpd:2.4.56-debian-ff

RUN apt-get update -y && \
    apt install -y libcjose0 libapache2-mod-auth-openidc libhiredis0.14 && \
    apt clean
RUN cp /usr/lib/apache2/modules/mod_auth_openidc.so /usr/local/apache2/modules/
```
However, httpd:2.4.56-debian-ff has a vulnerability. The fix version is 2.4.57-debian-ff.  But if I use 2.4.57-debian-ff as our httpd base docker image, our app will not start even though the installed without any error. So we had to go back to version httpd:2.4.56-debian-ff in order to make our app work again. 

Any suggestions? 

Thanks,
Fong

**Comment by michael-grunder** [comment]
Going to close this one as it's more an issue with an app that uses hiredis.
